modules/contact/src/constraints/ComputeFrictionalForceLMMechanicalContact.C

```diff
@@ -0,0 +1,215 @@
+      "ComputeFrictionalForceLMMechanicalContact relies on use of the global indexing container "
+      "in order to make its implementation feasible");
+#endif
+
+  if (!getParam<bool>("use_displaced_mesh"))
+    paramError("use_displaced_mesh",
+               "'use_displaced_mesh' must be true for the "
+               "ComputeFrictionalForceLMMechanicalContact object");
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::computeQpProperties()
+{
+  // Get the value of _qp_gap
+  ComputeWeightedGapLMMechanicalContact::computeQpProperties();
+
+  // Compute the value of weighted tangential velocity
+  // Build the velocity vector
+
+  if (_has_primary)
+  {
+    ADRealVectorValue relative_velocity(_secondary_x_dot[_qp] - _primary_x_dot[_qp],
+                                        _secondary_y_dot[_qp] - _primary_y_dot[_qp],
+                                        0);
+    // Add derivative information
+    relative_velocity(0).derivatives() =
+        _secondary_x_dot[_qp].derivatives() - _primary_x_dot[_qp].derivatives();
+    relative_velocity(1).derivatives() =
+        _secondary_y_dot[_qp].derivatives() - _primary_y_dot[_qp].derivatives();
+
+    // Get the component in the tangential direction
+    _qp_tangential_velocity = relative_velocity * _tangents[_qp][0];
+  }
+  else
+    _qp_tangential_velocity = std::numeric_limits<Real>::quiet_NaN();
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::computeQpIProperties()
+{
+  // Get the _node_to_weighted_gap map
+  ComputeWeightedGapLMMechanicalContact::computeQpIProperties();
+
+  const auto * const node = _lower_secondary_elem->node_ptr(_i);
+  _node_to_weighted_tangential_velocity[node] += _test[_i][_qp] * _qp_tangential_velocity;
+  //  Moose::out << "Node locations: x: " << _lower_secondary_elem->node_ptr(_i)->get_info() <<
+  //  "\n"; Moose::out << "_node_to_weighted_tangential_velocity[node].size(): "
+  //             << _node_to_weighted_tangential_velocity.size() << "\n";
+  //  Moose::out << "qp_tangential_vel: " << MetaPhysicL::raw_value(_qp_tangential_velocity) <<
+  //  "\n"; Moose::out << "_secondary_x_dot[_qp]: " << MetaPhysicL::raw_value(_secondary_x_dot[_qp])
+  //  << "\n";
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::residualSetup()
+{
+  // Clear _node_to_weighted_gap map
+  ComputeWeightedGapLMMechanicalContact::residualSetup();
+
+  _node_to_weighted_tangential_velocity.clear();
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::jacobianSetup()
+{
+  residualSetup();
+}
+
+void
+ComputeFrictionalForceLMMechanicalContact::computeResidual(const Moose::MortarType mortar_type)
+{
+  if (mortar_type != Moose::MortarType::Lower)
+    return;
+
+  // ComputeWeightedGapLMMechanicalContact::computeResidual(mortar_type);
+
+  // Fill node to weighted tangential velocity map.
+  for (_qp = 0; _qp < _qrule_msm->n_points(); _qp++)
+  {
+    computeQpProperties();
+    for (_i = 0; _i < _test.size(); ++_i)
+      computeQpIProperties();
+  }
+}
```

Comment:
I don't think this override is actually necessary right? This is what is already done in the base class?

Reply:
Oh good I thought I might be being stupid by just working on a single commit and i was right